### PR TITLE
3438 (docs details), 3472 (minor version)

### LINKED ISSUE #3438: Support promises for pipe and pipeWith
Currently pipe and pipeWith does not wait for promises to complete, enhance pipe functions to support promises.

### LINKED ISSUE #3472: 0.30.1 upgrade guide
[v0.30.0...v.0.30.1](https://github.com/ramda/ramda/compare/v0.30.0...v.0.30.1)

:new: Added:
https://github.com/ramda/ramda/pull/3442: ascendNatural and descendNatural

:boom: Breaking Changes:

:bulb: Changes:
https://github.com/ramda/ramda/pull/3463: revert placeholder update to support cross-version usage

:zap: Performance:

:clipboard: Documentation:

:heavy_check_mark:  Testing:
https://github.com/ramda/ramda/pull/3301: update version of fast-check

:rocket: Build and internal:

:star: Miscellaneous:

### PR DESCRIPTION
**Pull request by toni-sharpe**
Fixes #3438 

### refer to pipeWith from andThen docs with egs
* with examples to show the usage, two examples, one shows that a fn can easily be added to pipe, using  and the second shows the  being used independently, in both cases, the successful Promise resultis handled

### improve pipeWith example
* the example hadn't thought through, the edit should make it more sensible

Fixes #3472 

### actually did the upgrade, my master was only one minor version off

Wouldn't normally mix two

### CODE REVIEW

File: source/pipeWith.js [hunk 1]
```diff
@@ -21,8 +21,14 @@ import identity from './identity.js';
  * @param {Function} transformer The transforming function
  * @param {Array} functions The functions to pipe
  * @return {Function}
- * @see R.composeWith, R.pipe
- * @example
+ * @see R.andThen, R.composeWith, R.pipe
+ * @examples
+ *
+ *      const doubleFn = (req) => Promise.resolve(req * 2)
+ *      const x = R.pipeWith(
```
Comment (kedashoe) [inline source/pipeWith.js:28]: I believe the first argument to `pipeWith` should be `andThen`. Also we'll still have a promise here, so it's not really gonna result in just `16`. Maybe something like

```
const doubleFn = (req) => Promise.resolve(req * 2)
R.pipeWith(R.andThen, [doubleFn, R.inc])(8).then(console.log)
// logs 17
```

File: source/andThen.js [hunk 2]
```diff
@@ -16,20 +16,28 @@ import _assertPromise from './internal/_assertPromise.js';
...
  *
  *      //getMemberName :: String -> Promise ({ firstName, lastName })
  *      const getMemberName = R.pipe(
  *        makeQuery,
  *        fetchMember,
- *        R.andThen(R.pick(['firstName', 'lastName']))
+ *        R.andThen(pickName),
  *      );
  *
+ *      Alternately
```
Comment (kedashoe) [inline source/andThen.js:34]: Let's make this a comment
`// Alternately`

File: source/andThen.js [hunk 3]
```diff
@@ -16,20 +16,28 @@ import _assertPromise from './internal/_assertPromise.js';
...
+ *      const getMemberName = R.pipe(
+ *        makeQuery,
+ *        fetchMember,
+ *      )
+ *      R.pipeWith(R.andThen(pickName))(getMemberName)
```
Comment (kedashoe) [inline source/andThen.js:39]: Similar here, pipeWith expects a function and then a list of functions. Maybe

```
R.pipeWith(R.andThen, [getMemberName, pickName])('bob@gmail.com').then(console.log)
// logs {"firstName":"Bob","lastName":"Loblaw"}
```

**Review by kedashoe — CHANGES_REQUESTED** [review]
Sorry for the slow review here, ty for the PR @toni-sharpe

**Comment by toni-sharpe** [comment]
@kedashoe no problem,

Let me introduce myself, I've been using this library for some time, with a lot of success and really like it, it's in my high-bar tool-kit, it's used [here on my demo site](https://data-visualisation.pages.dev/) and you can find [the Github for that here](https://github.com/toni-sharpe/data-visualisation) - I will read the comments carefully and sort the problems out.

I would like to contribute to the library regularly as well as learning how it works so I will take care to understand the comments. I will have questions and need some guidance too but in return, hopefully, I'll solve lots of problems.

I'll take a look at "good first issues" next.

File: source/andThen.js (hunk 3 above)
Comment (toni-sharpe) [reply source/andThen.js:39]: @kedashoe so `pipeWith` is basically `pipe` but it uses the `fn` from the first arg (in the example `andThen` is called on `getMemberName` which produces a result, on which `andThen` is called again with `pickName`). Allowing the chainng of `then` in the Promise context.

File: source/pipeWith.js (hunk 1 above)
Comment (toni-sharpe) [reply source/pipeWith.js:28]: Again, my comment above explains my understanding and I'll use this example directly. Please confirm I understand.

**Comment by toni-sharpe** [comment]
@kedashoe I made the changes and understood more as a result, I hope, 

Also, `garden` is the best term, I got rid of `boy` at one employer from the term `boy-scouting` as well, some of us weren't boys, but scouting still has that military feel about it which is all a bit meh. We were deployed in "squads" there lol.

`garden` ~just blows it out~ ...... is a great improvement.

**Pull request merged**